app/controllers/collections_controller.rb

```diff
@@ -121,6 +121,60 @@ def revoke_access
     redirect_to licences_path
   end
 
+  def add_items_to_collection
+    # referenced documents (HCSVLAB-1019) are already handled by the look_for_documents part of the item ingest
+    begin
+      # Perform parameter transformations and request validations
+      params[:items] = JSON.parse(params[:items]) if params[:items].is_a? String
+      params[:file] = [params[:file]] unless params[:file].nil? or params[:file].is_a? Array
+      validate_add_items_request(params)
+
+      collection = Collection.find_by_name(params[:id])
+      corpus_dir = corpus_dir(params[:id])
+      items = []
+      items_added = []
+      uploaded_files = []
+
+      # Upload each of the files provided in the multipart HTTP request
+      if !params[:file].nil?
+        params[:file].each do |uploaded_file|
+          uploaded_files.push(upload_document_using_multipart(File.join(corpus_dir, uploaded_file.original_filename), uploaded_file, collection.name))
```

Comment:
You call `File.join(corpus_dir, uploaded_file.original_filename)` here, but he first line `upload_document_using_multipart` is `filename = File.basename absolute_filename`, it seems a bit funny to create a file path then unpack it again straight away. Maybe pass the corpus dir and the filename as separate params to `upload_document_using_multipart` and do the `File.join` there, that way you don't ever have to call `File.basename`